lib/get-assets.js

```diff
@@ -0,0 +1,96 @@
+const path = require('path');
```

Comment:
This code is different because always returns an Object and the github code returns String or object and validate again on every asset in publish.js: https://github.com/semantic-release/github/blob/master/lib/publish.js#L39, this code simplify that and has an extra null check https://github.com/semantic-release/gitlab/pull/55/files#diff-ef9cde994eea4bbb6446c1445f8dd8e8R18

Reply:
By the way the `typeof file === 'string'` you added doesn't seems to be useful as `files` will always be a list of Strings (either the return value of `dirGlob` or `castArray(getAssetPath(asset))`).

And mapping string definition asset to object in `get-assets` vs `publish` doesn't make much difference.

Reply:
that validation prevent when has an
![screenshot-2018-11-24_09-39-20](https://user-images.githubusercontent.com/14242544/48969463-ec340880-efcc-11e8-9140-b16e2d185b1d.png)
 invalid file

Reply:
@pvdlg copy/paste from github plugin: https://github.com/semantic-release/gitlab/pull/55/commits/d64e92db7f22d7e324fa1d1d8a43c17af7fa3fe8